### PR TITLE
12.6.6

### PR DESCRIPTION
**Pull request by JeanChristopheMorinPerso**
Channel: snowflake

Upstream: https://gitlab.com/conradsnicta/armadillo-code/-/tree/12.6.x?ref_type=heads
conda-forge: https://github.com/conda-forge/armadillo-feedstock

Depends on:
* https://github.com/AnacondaRecipes/superlu-feedstock/pull/1

### CODE REVIEW

**Comment by JeanChristopheMorinPerso** [comment]
Arg, I think I'll need to compile with OpenMP. Which means I'll need to figure out why I can't compile with OpenMP...

File: recipe/meta.yaml [hunk 1]
```diff
@@ -42,10 +49,16 @@ test:
     - if not exist %LIBRARY_LIB%\armadillo.lib exit 1  # [win]
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:54]: Will it build on Windows a static library, right? https://gitlab.com/conradsnicta/armadillo-code/-/blob/12.6.x/CMakeLists.txt?ref_type=heads#L181

**Review by skupr-anaconda — APPROVED** [review]
LGTM

File: recipe/meta.yaml (hunk 1 above)
Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml:49]: or is it just import library ?
https://learn.microsoft.com/en-us/windows/win32/dlls/dynamic-link-library-creation#creating-an-import-library

File: recipe/meta.yaml [hunk 2]
```diff
@@ -6,14 +6,18 @@ package:
 
 source:
   fn: armadillo-{{ version }}.tar.xz
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/meta.yaml]: not needed

File: recipe/meta.yaml [hunk 3]
```diff
@@ -6,14 +6,18 @@ package:
...
-  url: http://sourceforge.net/projects/arma/files/armadillo-{{ version }}.tar.xz
+  url: https://sourceforge.net/projects/arma/files/armadillo-{{ version }}.tar.xz
   sha256: 3858b0fd431772af032ad3f35c2aeb54e8dabea59169e7d1e9fccd78bc82ad35
 
 build:
   number: 0
   skip: true  # [win and vc<14]
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/meta.yaml:14]: so we will have windows with vc > 14, but there is no bld.bat ?

File: recipe/meta.yaml (hunk 1 above)
Comment (lorepirri) [reply recipe/meta.yaml:54]: Good catch. I can only see `armadillo.lib` in the log for `win` https://cloud.prefect.io/anaconda/flow-run/f2d6d989-07e1-4972-af3b-42cf95f92b37

also: 
https://gitlab.com/conradsnicta/armadillo-code/-/blob/12.6.x/CMakeLists.txt?ref_type=heads#L223-228

```
if(MSVC AND BUILD_SHARED_LIBS)
  message(STATUS "")
  message(STATUS "*** WARNING: building shared library with MSVC is not supported." )
  message(STATUS "*** WARNING: if anything breaks, you get to keep all the pieces.")
  message(STATUS "")
endif()
```

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:54]: Yes, it defaults to static and I don't want to build in shared because of that warning.

**Comment by lorepirri** [comment]
> Arg, I think I'll need to compile with OpenMP. Which means I'll need to figure out why I can't compile with OpenMP...

Is it some similar trick for llvm-openmp / intel-openmp conflict?

https://github.com/AnacondaRecipes/pytorch-forecasting-feedstock/blob/d8f6e07bc093bb1a0c012669a62f4177c5bce2e2/recipe/meta.yaml#L25-L35

**Review by lorepirri — APPROVED** [review]

**Comment by lorepirri** [comment]
[PKG-3197]

[PKG-3197]: https://anaconda.atlassian.net/browse/PKG-3197?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Pull request merged**